astroid/inference_tip.py

```diff
@@ -31,16 +36,22 @@ def _inference_tip_cached(
 
     def inner(*args: _P.args, **kwargs: _P.kwargs) -> Iterator[InferenceResult]:
         node = args[0]
-        try:
-            result = _cache[func, node]
+        context = args[1]
+        partial_cache_key = (func, node)
+        if partial_cache_key in _CURRENTLY_INFERRING:
```

Comment:
Why is this a partial key?

Reply:
Using a full key caused the recursion error on PyPy. It's a pragmatic compromise to avoid so much recursive inference with slightly different contexts while still passing the simple test cases from the linked issues.

Reply:
I think it would be nice to add a comment explaining this. It is not really clear from the code why we do this.